tests/test_single_topology_v3.py

```diff
@@ -392,98 +393,83 @@ def test_no_chiral_bond_restraints():
     _ = U(init_conf)
 
 
-def assert_interpolation_valid(f, a, b):
-    np.testing.assert_allclose(f(a, b, 0.0), a)
-    np.testing.assert_allclose(f(a, b, 1.0), b)
+finite_floats = functools.partial(st.floats, allow_nan=False, allow_infinity=False, allow_subnormal=False)
 
-    rng = np.random.default_rng(2022)
-    lambdas = rng.uniform(0, 1, (30,))
-    ys = vmap(functools.partial(f, a, b))(lambdas)
-    assert np.all(np.minimum(a, b) <= ys)
-    assert np.all(ys <= np.maximum(a, b))
+nonzero_force_constants = finite_floats(1e-9, 1e9)
 
+lambdas = finite_floats(0.0, 1.0)
 
-def test_handle_ring_opening_closing():
-    rng = np.random.default_rng(2022)
-    n = 100
-    k = 20
+open_lambda_intervals = st.lists(finite_floats(1e-9, 1.0 - 1e-9), min_size=2, max_size=2, unique=True).map(sorted)
 
-    def random_ks():
-        return rng.uniform(1, 2, (n,))
 
-    src_k = random_ks()
-    dst_k = random_ks()
-
-    idxs = rng.choice(n, k, replace=False)
-
-    closing_idxs = idxs[: k // 2]
-    opening_idxs = idxs[k // 2 :]
-
-    src_k[closing_idxs] = 0.0
-    dst_k[opening_idxs] = 0.0
-
-    lambda_min = 0.3
-    lambda_max = 0.7
+@given(nonzero_force_constants, open_lambda_intervals, lambdas)
```

Comment:
https://hypothesis.readthedocs.io/en/latest/details.html#hypothesis.given says that this is a `randomized` test. Is it deterministic randomness?

Reply:
Do we need to do something like this? https://hypothesis.readthedocs.io/en/latest/details.html?highlight=random#making-random-code-deterministic

Reply:
Or perhaps the more explicit https://hypothesis.readthedocs.io/en/latest/reproducing.html?highlight=random#reproducing-a-test-run-with-seed?

Reply:
Yes, good call, I think it makes sense to explicitly use `@seed`, otherwise it seems like we might see nondeterministic (if reproducible) test failures. It looks like [`derandomize`](https://hypothesis.readthedocs.io/en/latest/settings.html#hypothesis.settings.derandomize) might be another option

Reply:
`derandomize` seems to depend on the actual function hash though, which seems dangerous. `@seed` makes things the most explicit.

Reply:
> Do we need to do something like this?

My understanding is that this is only needed for tests that have sources of nondeterminism independent of the example generation, so I don't think it's needed.